CHANGELOG.md

```diff
@@ -3,6 +3,26 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/#semantic-versioning-200).
 
+## [2.3.0] - 2023-11-23
+### :magic_wand: Added
+- Endpoint override for secrets manager plugin ([PR #707](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/707)).
+- Allow users to set up a lambda to initialize new connections ([PR #705](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/705)).
+- Introduced Dialect.prepareConnectProperties ([PR #704](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/704)).
+- Native telemetry support in the AWS JDBC Driver ([PR #617](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/617)).
+
+### :bug: Fixed
+- Continue monitoring if unhandled Exception is thrown ([PR #676](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/676)).
+- Password properties are now masked in logs ([PR #701](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/701)).
+- Issue when getting a connection for a closed statement ([PR #682](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/682)).
+- Maven coordinates in README ([PR #681](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/681)).
+
+### :crab: Changed
+- Added buffer to iam token expiry and create token expiry time earlier ([PR #706](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/706)).
```

Comment:
Could we clarify earlier here? I've suggested removing it since it's not too clear.

```suggestion
- Added buffer to IAM token expiry and create token expiry time ([PR #706](https://github.com/awslabs/aws-advanced-jdbc-wrapper/pull/706)).
```